node/src/bead/tests.rs

```diff
@@ -274,3 +276,114 @@ fn test_codec_response_roundtrip() {
+        vec![BeadHash::from_byte_array([0u8; 32])]
+            .into_iter()
+            .collect(),
+    );
+
+    // Serialize
+    let mut buffer = Vec::new();
+    request.consensus_encode(&mut buffer).unwrap();
+    let io = Cursor::new(buffer);
+
+    // Deserialize
+    let protocol = libp2p::StreamProtocol::new("/braidpool/1.0.0");
+    let decoded_request =
+        block_on(codec.read_request(&protocol, &mut futures::io::AllowStdIo::new(io))).unwrap();
+    assert_eq!(request, decoded_request);
+}
+
+// test codec for all bead request types
+#[test]
+fn test_bead_request_codec() {
+    let mut codec = BeadCodec::default();
+
+    for request in vec![
+        BeadRequest::GetTips,
+        BeadRequest::GetBeads(Vec::from([BeadHash::from_byte_array([0u8; 32])])),
+        BeadRequest::GetGenesis,
+        BeadRequest::GetAllBeads,
+        BeadRequest::GetBeadsAfter(vec![BeadHash::from_byte_array([0u8; 32])]),
+    ] {
+        // Serialize
+        let mut buffer = Vec::new();
+        request.consensus_encode(&mut buffer).unwrap();
+        let io = Cursor::new(buffer);
+
+        // Deserialize
+        let protocol = libp2p::StreamProtocol::new("/braidpool/1.0.0");
+        let decoded_request =
+            block_on(codec.read_request(&protocol, &mut futures::io::AllowStdIo::new(io))).unwrap();
+        assert_eq!(request, decoded_request);
+    }
+}
+
+#[test]
+fn test_bead_response_codec() {
+    let mut codec = BeadCodec::default();
+
+    // Create test bead for responses that need beads
+    let test_bead = create_test_bead(1, None);
+    let test_hash = BeadHash::from_byte_array([1u8; 32]);
+    let test_hash2 = BeadHash::from_byte_array([2u8; 32]);
+
+    // Test all BeadResponse variants
+    let responses = vec![
+        BeadResponse::Beads(vec![test_bead.clone()]),
+        BeadResponse::Tips(vec![test_hash, test_hash2]),
+        BeadResponse::Genesis(vec![test_hash]),
+        BeadResponse::GetAllBeads(vec![test_bead.clone(), test_bead.clone()]),
+        BeadResponse::GetBeadsAfter(vec![test_bead.block_header.block_hash()]),
+        BeadResponse::Error(BeadSyncError::GenesisMismatch),
+        BeadResponse::Error(BeadSyncError::Other("Test error message".to_string())),
+    ];
+
+    for response in responses {
+        // Serialize
+        let mut buffer = Vec::new();
+        response.consensus_encode(&mut buffer).unwrap();
+        let io = Cursor::new(buffer);
+
+        // Deserialize
+        let protocol = libp2p::StreamProtocol::new("/braidpool/1.0.0");
+        let decoded_response =
+            block_on(codec.read_response(&protocol, &mut futures::io::AllowStdIo::new(io)))
+                .unwrap();
+        assert_eq!(response, decoded_response);
+    }
+}
+
+#[test]
+fn test_bead_sync_error_codec() {
+    // Test BeadSyncError encoding/decoding directly (not through codec)
+    let errors = vec![
+        BeadSyncError::GenesisMismatch,
+        BeadSyncError::Other("Network timeout".to_string()),
```

Comment:
Remove the `BeadSyncError::Other` cases. Network timeout is handled by libp2p which will fire a `Event::OutboundFailure` so we don't need to handle this in BeadSyncError and we'd never be able to decode such a message anyway because it hits lip2p first.